lib/codecs/src/decoding/framing/character_delimited.rs

```diff
@@ -38,6 +38,13 @@ pub struct CharacterDelimitedDecoderOptions {
     /// The maximum length of the byte buffer.
     ///
     /// This length does *not* include the trailing delimiter.
+    ///
+    /// By default, there is no maximum length enforced. If events are malformed, this can lead to
+    /// additional resource usage as events continued to be buffered in memory, and can potentially
+    /// lead to memory exhaustion in extreme cases.
+    ///
+    /// Consider setting this to a reasonable value, based on the expected maximum buffer length, if
+    /// there is a risk of malformed data, such as if user-controlled data is consumed.
```

Comment:
💭 I feel like if the clauses were switched here it would read easier, unless I don't understand the sentence.

```suggestion
    ///  If there is a risk of malformed data, such as if user-controlled data is consumed,
    /// consider setting this to a reasonable value, based on the expected maximum buffer length.
```

Reply:
Yeah, I'm with you: start out by mentioning the conditional first, and then explain the solution.

I took your structural change, and most of the wording, and tweaked it ever so slightly. It occurred to me that "reasonable value" and "expected maximum buffer length" mean almost nothing if the whole point is that you have some input that you're not entirely sure how large it will be.